### DISCUSSION TITLE
DPoPWebIdExtractor jwk thumprint verification

### DISCUSSION THREAD

**Discussion by zg009** [panva/jose#722] [Q&A] [answered]
Hello,

I am trying to register for websocket notifications using CS v6.x, and am authenticating successfully using the provided authentication documents. However, when I try to authenticate with the Websocket endpoint (/.notifications/WebSocketChannel2023) I receive the error ```Expected object property cnf``` in the server logs. I know how to calculate the thumbprint, I am confused about which key I need to get the thumbprint of and where to put it - for an asymmetric keypair, does it need to be the public key? Also, how am I supposed to structure it? I am believing 
```
cnf: {
     jkt: <thumbprint calculation here>
     }
```
Is correct. Also, does it expect any hashing using SHA-1? Thanks

**Comment by panva** [comment, marked as answer]
https://datatracker.ietf.org/doc/html/rfc9449#section-6.1

> jkt: JWK SHA-256 Thumbprint confirmation method. The value of the jkt member MUST be the base64url encoding (as defined in [[RFC7515](https://datatracker.ietf.org/doc/html/rfc7515)]) of the JWK SHA-256 Thumbprint (according to [[RFC7638](https://datatracker.ietf.org/doc/html/rfc7638)]) of the DPoP public key (in JWK format) to which the access token is bound.


There's a helper for this in `jose`: https://github.com/panva/jose/blob/v5.9.3/docs/functions/jwk_thumbprint.calculateJwkThumbprint.md#function-calculatejwkthumbprint

**Reply by zg009** [reply]
I am not working in a Javascript environment, but thank you for the resource. I was able to resolve my issue.

**Reply by panva** [reply]
Then I don't understand what lead you to this project's discussions. But anyway, glad you resolved your issue.

**Reply by zg009** [reply]
I believe I meant to post this in the CommunitySolidServer github as they used an API which depends upon this library - I am working with their server from a service written in a separate language (Java), and was receiving the server-side error. Thank you for the help though 👍
